packages/devtools_app/lib/src/memory/memory_tracker_model.dart

```diff
@@ -0,0 +1,259 @@
+  String getValue(Tracker dataObject) {
+    if (dataObject is TrackerClass) {
+      return '${dataObject.children.length}';
+    }
+
+    assert(dataObject is TrackerAllocation ||
+        dataObject is TrackerCall ||
+        dataObject is TrackerMore);
+    return '';
+  }
+
+  @override
+  double get fixedWidthPx => 50.0;
+
+  @override
+  bool get supportsSorting => true;
+}
+
+class TreeTracker {
+  TrackerClass tree1;
+
+  TreeColumnData<Tracker> treeColumn = _TrackerClassColumn();
+
+  final selectionNotifier =
+      ValueNotifier<Selection<Tracker>>(Selection<Tracker>());
+
+  /// Rebuild list of classes to track.
+  void createTrackerTree(
+    Map<String, ClassRef> classesTracked,
+    List<AllocationSamples> allocationSamples,
+  ) {
+    tree1 = TrackerClass('_ROOT_tracking');
+
+    tree1.children.clear();
+
+    final classesName = classesTracked.keys;
+    for (var className in classesName) {
+      final classEntry = TrackerClass(className);
+      tree1.addChild(classEntry);
+
+      final trackedClassId = classesTracked[className].id;
+
+      final allocationSample = allocationSamples.singleWhere((sample) {
+        // If no stacktraces, then nothing to handle.
+        if (sample.totalStacktraces == 0) return false;
+
+        // Match the sample to the class being tracked.
+        final allocationClassRef = sample.classRef;
+        final allocationClassId = allocationClassRef.id;
+        return allocationClassId == trackedClassId;
+      }, orElse: () => null);
+
+      if (allocationSample != null) {
+        for (var stacktrace in allocationSample.stacktraces) {
+          final allocation = TrackerAllocation(stacktrace.timestamp);
+          classEntry.addChild(allocation);
+          int displayDepth = 0;
+          for (var callStackEntry in stacktrace.stacktrace) {
+            if (allocation.displayAll ||
+                displayDepth++ < allocation.callDepthDisplay) {
+              allocation.addChild(TrackerCall(callStackEntry));
+            }
+          }
+          if (!allocation.displayAll) {
+            allocation.addChild(TrackerMore(stacktrace.stacktrace));
+          }
+        }
+      }
+    }
+  }
+
+  /// Remove the "More ..." and show the entire call stack.
+  void expandCallStack(TrackerMore item) {
+    final TrackerAllocation allocation = item.parent;
+
+    // Show the entire call stack for this instance.
+    allocation.displayAll = true;
+
+    // Re-construct the callstack for this instance.
+    final leafNodesCount = allocation.children.length;
+
+    assert(allocation.children.last is TrackerMore);
+    allocation.removeLastChild();   // Remove "More ..."
+
+    // Add the entire call stack as leaf nodes.
+    final totalCallStack = item.stacktrace.length;
+    for (var index = leafNodesCount - 1; index < totalCallStack; index++) {
+      allocation.addChild(TrackerCall(item.stacktrace[index]));
+    }
+    allocation.expandCascading();
+  }
+
+  void createTrackerTreeSample() {
```

Comment:
this looks like test code move it to a test directory file.

Reply:
Done.